tests/src/test/scala/org/http4s/internal/TlsSuite.scala

```diff
@@ -0,0 +1,381 @@
+    "TLS_DHE_PSK_WITH_CAMELLIA_256_GCM_SHA384" -> 256,
+    "TLS_RSA_PSK_WITH_CAMELLIA_128_GCM_SHA256" -> 128,
+    "TLS_RSA_PSK_WITH_CAMELLIA_256_GCM_SHA384" -> 256,
+    "TLS_PSK_WITH_CAMELLIA_128_CBC_SHA256" -> 128,
+    "TLS_PSK_WITH_CAMELLIA_256_CBC_SHA384" -> 256,
+    "TLS_DHE_PSK_WITH_CAMELLIA_128_CBC_SHA256" -> 128,
+    "TLS_DHE_PSK_WITH_CAMELLIA_256_CBC_SHA384" -> 256,
+    "TLS_RSA_PSK_WITH_CAMELLIA_128_CBC_SHA256" -> 128,
+    "TLS_RSA_PSK_WITH_CAMELLIA_256_CBC_SHA384" -> 256,
+    "TLS_ECDHE_PSK_WITH_CAMELLIA_128_CBC_SHA256" -> 128,
+    "TLS_ECDHE_PSK_WITH_CAMELLIA_256_CBC_SHA384" -> 256,
+    "TLS_RSA_WITH_AES_128_CCM" -> 128,
+    "TLS_RSA_WITH_AES_256_CCM" -> 256,
+    "TLS_DHE_RSA_WITH_AES_128_CCM" -> 128,
+    "TLS_DHE_RSA_WITH_AES_256_CCM" -> 256,
+    "TLS_RSA_WITH_AES_128_CCM_8" -> 128,
+    "TLS_RSA_WITH_AES_256_CCM_8" -> 256,
+    "TLS_DHE_RSA_WITH_AES_128_CCM_8" -> 128,
+    "TLS_DHE_RSA_WITH_AES_256_CCM_8" -> 256,
+    "TLS_PSK_WITH_AES_128_CCM" -> 128,
+    "TLS_PSK_WITH_AES_256_CCM" -> 256,
+    "TLS_DHE_PSK_WITH_AES_128_CCM" -> 128,
+    "TLS_DHE_PSK_WITH_AES_256_CCM" -> 256,
+    "TLS_PSK_WITH_AES_128_CCM_8" -> 128,
+    "TLS_PSK_WITH_AES_256_CCM_8" -> 256,
+    "TLS_PSK_DHE_WITH_AES_128_CCM_8" -> 128,
+    "TLS_PSK_DHE_WITH_AES_256_CCM_8" -> 256,
+    "TLS_ECDHE_ECDSA_WITH_AES_128_CCM" -> 128,
+    "TLS_ECDHE_ECDSA_WITH_AES_256_CCM" -> 256,
+    "TLS_ECDHE_ECDSA_WITH_AES_128_CCM_8" -> 128,
+    "TLS_ECDHE_ECDSA_WITH_AES_256_CCM_8" -> 256,
+    "TLS_ECCPWD_WITH_AES_128_GCM_SHA256" -> 128,
+    "TLS_ECCPWD_WITH_AES_256_GCM_SHA384" -> 256,
+    "TLS_ECCPWD_WITH_AES_128_CCM_SHA256" -> 128,
+    "TLS_ECCPWD_WITH_AES_256_CCM_SHA384" -> 256,
+    "TLS_SHA256_SHA256" -> 0,
+    "TLS_SHA384_SHA384" -> 0,
+    "TLS_GOSTR341112_256_WITH_KUZNYECHIK_CTR_OMAC" -> 256,
+    "TLS_GOSTR341112_256_WITH_MAGMA_CTR_OMAC" -> 256,
+    "TLS_GOSTR341112_256_WITH_28147_CNT_IMIT" -> 256,
+    "TLS_GOSTR341112_256_WITH_KUZNYECHIK_MGM_L" -> 256,
+    "TLS_GOSTR341112_256_WITH_MAGMA_MGM_L" -> 256,
+    "TLS_GOSTR341112_256_WITH_KUZNYECHIK_MGM_S" -> 256,
+    "TLS_GOSTR341112_256_WITH_MAGMA_MGM_S" -> 256,
+    "TLS_ECDHE_RSA_WITH_CHACHA20_POLY1305_SHA256" -> 256,
+    "TLS_ECDHE_ECDSA_WITH_CHACHA20_POLY1305_SHA256" -> 256,
+    "TLS_DHE_RSA_WITH_CHACHA20_POLY1305_SHA256" -> 256,
+    "TLS_PSK_WITH_CHACHA20_POLY1305_SHA256" -> 256,
+    "TLS_ECDHE_PSK_WITH_CHACHA20_POLY1305_SHA256" -> 256,
+    "TLS_DHE_PSK_WITH_CHACHA20_POLY1305_SHA256" -> 256,
+    "TLS_RSA_PSK_WITH_CHACHA20_POLY1305_SHA256" -> 256,
+    "TLS_ECDHE_PSK_WITH_AES_128_GCM_SHA256" -> 128,
+    "TLS_ECDHE_PSK_WITH_AES_256_GCM_SHA384" -> 256,
+    "TLS_ECDHE_PSK_WITH_AES_128_CCM_8_SHA256" -> 128,
+    "TLS_ECDHE_PSK_WITH_AES_128_CCM_SHA256" -> 128
+  )
+
+  test("deduceKeyLength should return expected results for known cipher suites") {
+    cipherSuiteMap.foreach { case (name, expectedValue) =>
+      assertEquals(tls.deduceKeyLength(name), expectedValue)
+    }
+  }
```

Comment:
Just a minor suggestion: what if we do something like this, so that if a test fails its easy to see which cipher it was.
```suggestion
  cipherSuiteMap.foreach { case (name, expectedValue) =>
    test(s"deduceKeyLength should return expected result for $name") {
      assertEquals(tls.deduceKeyLength(name), expectedValue)
    }
  }
```